core/domain/acl_decorators.py

```diff
@@ -126,3 +167,320 @@ def test_can_view_stats(self, exploration_id, **kwargs):
+        else:
+            raise self.UnauthorizedUserException(
+                'User %s does not have permissions to delete exploration %s' %
+                (self.user_id, exploration_id))
+
+    return test_can_delete
+
+
+def can_suggest_changes_to_exploration(handler):
+    """"Decorator to check whether a user can make suggestions to an
+    exploration.
+    """
+    def test_can_suggest(self, exploration_id, **kwargs):
+        if (
+                role_services.ACTION_SUGGEST_CHANGES_TO_EXPLORATION in
+                self.actions and user_services.has_fully_registered(
+                    self.user_id)):
+            return handler(self, exploration_id, **kwargs)
+        else:
+            raise self.UnauthorizedUserException(
+                'You do not have credentials to give suggestions to ' +
+                'exploration.')
+
+    return test_can_suggest
+
+
+def can_manage_suggestions_on_exploration(handler):
+    """Decorator to check whether a user can act upon the suggestions made
+    by learners."""
+
+    def test_can_manage(self, exploration_id, **kwargs):
+        if not self.user_id:
+            self.redirect(current_user_services.create_login_url(
+                self.request.uri))
+            return
+        exploration_rights = rights_manager.get_exploration_rights(
+            exploration_id)
+
+        if exploration_rights is None:
+            raise self.PageNotFoundException
+
+        if check_exploration_editable(
+                self.user_id, self.actions, exploration_rights):
+            return handler(self, exploration_id, **kwargs)
+        else:
+            raise self.UnauthorizedUserException(
+                'You do not have the credentials to manage suggestions for ' +
+                'this exploration.')
+
+    return test_can_manage
+
+
+def can_publish_exploration(handler):
+    """Decorator to check whether user can publish exploration."""
+
+    def test_can_publish(self, exploration_id, **kwargs):
+        exploration_rights = rights_manager.get_exploration_rights(
+            exploration_id)
+
+        if exploration_rights is None:
+            raise self.PageNotFoundException
+
+        if exploration_rights.cloned_from:
+            raise self.UnauthorizedUserException(
+                'You do not have credentials to publish this exploration.')
+
+        if role_services.ACTION_PUBLISH_ANY_EXPLORATION in self.actions:
+            return handler(self, exploration_id, **kwargs)
+
+        if exploration_rights.status == rights_manager.ACTIVITY_STATUS_PRIVATE:
+            if role_services.ACTION_PUBLISH_OWNED_EXPLORATION in self.actions:
+                if self.user_id in exploration_rights.owner_ids:
+                    return handler(self, exploration_id, **kwargs)
+
+        if exploration_rights.status == rights_manager.ACTIVITY_STATUS_PUBLIC:
+            if role_services.ACTION_PUBLISH_PUBLIC_EXPLORATION in self.actions:
```

Comment:
"publicize" is different from "publish", sorry about the confusion. There are currently three stages: private --> public --> publicize. "Publish" is the first transition and "publicize" is the second. We are actually planning to get rid of the "publicize" status but for the purposes of this PR I think we'll just need to deal with it as a separate action.